orchestrator-salt/src/main/resources/salt/salt/nginx/conf/clouderamanager-ssl-user-facing.conf

```diff
@@ -3,34 +3,47 @@ map $http_upgrade $connection_upgrade {
         ''      close;
     }
 
+upstream knox {
+    server 127.0.0.1:8443;
+}
+
 server {
     listen       443;
     ssl on;
     ssl_certificate      /etc/certs-user-facing/server.pem;
     ssl_certificate_key  /etc/certs-user-facing/server-key.pem;
-    rewrite ^([^/]*/clouderamanager)$ $1/ permanent;
-    # e.g.: https://172.22.107.133/img/white-logo.png -> https://172.22.107.133/clouderamanager/img/white-logo.png
-    if ($http_referer ~ .*/clouderamanager/.*) {
-        rewrite ^([/](?!clouderamanager/).*$) /clouderamanager$1;
-    }
+
+    #If hadoop-jwt token is defined do not rewrite the path to go to clouderamanager -> go to knox
     if ($cookie_CLOUDERA_MANAGER_SESSIONID ~ .+) {
-        rewrite ^([/](?!clouderamanager/).*$) /clouderamanager$1;
+        set $rewritecondition  "cm-cookie";
     }
     if ($cookie_SESSION ~ .+) {
-        rewrite ^([/](?!clouderamanager/).*$) /clouderamanager$1;
+        set $rewritecondition  "cm-cookie";
     }
-    location / {
-        rewrite ^(/)$  https://$host/clouderamanager;
+    if ($cookie_hadoop-jwt ~ .+) {
+        set $rewritecondition  "${rewritecondition}+hadoop-jwt-cookie";
     }
+    if ($rewritecondition = "cm-cookie") {
+        rewrite ^([/](?!clouderamanager/).*$) /clouderamanager$1;
+    }
+
     location ~ .*/clouderamanager/(.*) {
-        proxy_pass         {{ protocol }}://clouderamanager/$1$is_args$args;
+        proxy_pass         https://clouderamanager/$1$is_args$args;
         proxy_redirect     http:// https://;
         proxy_http_version 1.1;
         proxy_set_header   Host $host;
         proxy_set_header   X-Forwarded-Host $server_name;
         proxy_set_header   X-Forwarded-Proto $scheme;
         proxy_set_header   Upgrade $http_upgrade;
         proxy_set_header   Connection $connection_upgrade;
-        proxy_set_header   Referer {{ protocol }}://$host/$1;
+        proxy_set_header   Referer https://$host/$1;
+    }
+    location ~ .*/ {
+        proxy_pass         https://knox;
+        proxy_redirect     off;
+        proxy_set_header   Host $host;
+        proxy_set_header   Referer https://$host/;
+        proxy_set_header   X-Forwarded-Host $server_name;
+        proxy_set_header   X-Forwarded-Proto $scheme;
     }
```

Comment:
do we really want to redirect everything to knox in every case? like what happens if we choose not to install knox?

Reply:
I mark this one as resolved because CB does redirect to the CM UI's direct route if gateway/Knox is not enabled.
https://github.com/hortonworks/cloudbreak/pull/6444/files#diff-6e00cc8735c353460aa0fd4845162ea4R78

Reply:
Right, it had caused some trouble if both are configured.  So the only thing what we can do is to redirect everything to knox, and if something fails and you would like to debug then you need to touch the nginx conf.

@biharitomi maybe some comment or guidance in the file would be nice, which says that if knox has failed then please uncomment these few lines and we will bypass knox... This would be super useful for debugging scenarios.